### PR TITLE
DevEx: use docketNumber instead of caseId for dynamo keys

### PR DESCRIPTION
**Pull request by rachaelparris**

### CODE REVIEW

File: shared/src/persistence/dynamo/cases/getCaseByDocketNumber.js [hunk 1]
```diff
@@ -13,29 +13,20 @@ exports.getCaseByDocketNumber = async ({
...
     ExpressionAttributeValues: {
-      ':pk': `case-by-docket-number|${docketNumber}`,
+      ':pk': `case|${docketNumber}`,
     },
     KeyConditionExpression: '#pk = :pk',
     applicationContext,
   });
 
-  if (results.length === 0) {
+  if (!caseItems) {
```
Comment (kkoskelin) [inline shared/src/persistence/dynamo/cases/getCaseByDocketNumber.js:27]: Was the old code a dangerous assumption that `results` would always be an array?

Comment (rachaelparris) [reply shared/src/persistence/dynamo/cases/getCaseByDocketNumber.js:27]: I just copy/pasted from `getCaseByCaseId` and updated to use `docketNumber` instead. This is what `getCaseByCaseId` was doing before - should it be checking for `length === 0`?

**Review by kkoskelin — APPROVED** [review]

File: shared/src/persistence/dynamo/cases/getCaseByDocketNumber.js (hunk 1 above)
Comment (kkoskelin) [reply shared/src/persistence/dynamo/cases/getCaseByDocketNumber.js:27]: I'm unsure -- I didn't try to pull all of it apart, just noticed the change in the kind of condition we were checking. I was wondering whether `client.query()` ever returns something falsy (knowing that an empty array is truthy).

Comment (rachaelparris) [reply shared/src/persistence/dynamo/cases/getCaseByDocketNumber.js:27]: I'll keep this in mind during testing when we've completed this refactor, and if we run into issues I'll come back and fix this.

**Pull request merged**